### ISSUE TITLE
We could use a Framework-wide standard DateTime filter

### ISSUE THREAD

**Issue by TheNando** [rackerlabs/encore-ui#1634]
Datetimes are used all over the place and every app seems to have it's own pattern. A framework-wide directive would help standardize display of dates and reduce testing scope. This pre-existing directive in Phone-UI would be a good starting point:
https://github.com/rackerlabs/encore-phone-ui/blob/master/app/src/common/filters/encoreDateTime.js

I imagine it could even have more functionality, such as a param to select a short, medium, or full format.

**Comment by CITguy** [comment]
I'd move to supplying filters rather than directives. Then we can provide functionality to supply predefined formats for (potentially) "long", "short", etc. Something like the following...

### Date
```html
{{ myDate | rxDate }} <!-- defaults to "long" format -->
{{ myDate | rxDate:'long' }}
{{ myDate | rxDate:'short' }}
```

### Time
```html
{{ myTime | rxTime }} <!-- defaults to "long" format -->
{{ myTime | rxTime:'long' }}
{{ myTime | rxTime:'short' }}
```

### Date + Time
```html
{{ myDateTime | rxDateTime }} <!-- defaults to "long" format -->
{{ myDateTime | rxDateTime:'long' }}
{{ myDateTime | rxDateTime:'short' }}
```

**Comment by CITguy** [comment]
Captured in JIRA [FRMW-817](https://jira.rax.io/browse/FRMW-817)

**Comment by TheNando** [comment]
Yes. I totally meant filter.

**Comment by ErnieAtLYD** [comment]
+1 for this. As a context, Nando and I are both on billing, where the long format is suited for things like server longs. In our cases, dates are used for support racker references to see when credit cards have been added or expired.

**Comment by 100stacks** [comment]
@TheNando @ErnieAtLYD: @kevincareer427 started working on this as a WIP in PR https://github.com/rackerlabs/encore-ui/pull/1755.

Please take a look for any feedback.  Thanks!

**Comment by CITguy** [comment]
@TheNando, with `encore-ui 2.1.0` we now have `rxDate`, `rxDateTime`, `rxMonth` and `rxTime` filters at your disposal. If you need the official Moment.js formats elsewhere, the `rxMomentFormats` value is also available.

**Comment by CITguy** [comment]
Fixed in 2.1.0
